prov/verbs/src/verbs_ep.c

```diff
@@ -641,7 +641,10 @@ int fi_ibv_open_ep(struct fid_domain *domain, struct fi_info *info,
 		}
 
 		ep->id->context = &ep->util_ep.ep_fid.fid;
-		ep->util_ep.ep_fid.msg = &fi_ibv_msg_ep_msg_ops;
+		if (dom->util_domain.threading == FI_THREAD_DOMAIN)
```

Comment:
Can this be relaxed to `dom->util_domain.threading >= FI_THREAD_ENDPOINT`? If the application can guarantee that access to TX interfaces of an endpoint would be serialized, we should be able to use the static variables without any races.